### PR TITLE
improve/simplify promisc handling

### PR DESCRIPTION
**Pull request by vsamel**
This series cleans up and improves promiscuous mode handling:
- enables it only on running interfaces
- hides it by move to capt_* abstraction
- changes it to the recommended way of enabling

### CODE REVIEW

File: src/promisc.c [hunk 1]
```diff
@@ -14,24 +14,71 @@ promisc.c	- handles the promiscuous mode flag for the Ethernet/FDDI/
...
+		return;
+	}
+	struct promisc_list *new = xmallocz(sizeof(*new));
+
+	new->ifindex = ifindex;
+	strcpy(new->ifname, dev);
+	list_add_tail(&new->list, promisc);
 
-		promisc_add_dev(promisc, device_name);
+	return;
```
Comment (pecameron) [inline src/promisc.c]: This should not be needed.

**Review by pecameron — APPROVED** [review]
I am not all that familiar with the code, so I am not sure how much I contributed in the review. It looks good to me with a minor nit.

File: src/promisc.c (hunk 1 above)
Comment (vsamel) [reply src/promisc.c]: Fixed.

**Comment by pecameron** [comment]
@vsamel Could you add a description at the top? Maybe from the commit that follow?

**Comment by pecameron** [comment]
@vsamel Is there anything else to do here before merging (after adding the description)?

**Comment by vsamel** [comment]
> @vsamel Could you add a description at the top? Maybe from the commit that follow?

  Done.  

> @vsamel Is there anything else to do here before merging (after adding the description)?

  That's all, you can merge it.

**Pull request merged**